File: Leet Code/romanToInteger.js
/**
 * @param {string} s
 * @return {number}
 */
 var romanToInt = function(s) {
  let numerals = {  // set values for each roman numeral
      I: 1,
      V: 5, 
      X: 10,
      L: 50,
      C: 100,
      D: 500,
      M: 1000
  }
  let sum = 0 // start count at 0
  
  for (let i = 0; i < s.length; i++) { // iterate through
    
    newNumber = numerals[s[i]] // set default to numeral value
    
    if (numerals[s[i]] < numerals[s[i + 1]]) { // if two values need subtracted
      newNumber = numerals[s[i + 1]] - numerals[s[i]] // set number to subtracted value
      i++ // skip to next number that needs added
    } 
    
    sum = sum + newNumber // add the newNumber to the total so far
  }
  return sum // return the total value
};

// romanToInt('III') //=> 3
// romanToInt('IV') // => 4
// romanToInt('XXXVII') // => 37
// romanToInt('XL') // => 40
// romanToInt('XCVIII') // => 98
// romanToInt('MCCXLIX') // => 1,249

File: Leet Code/romanToInteger.rb
# @param {String} s
# @return {Integer}
def roman_to_int(s)
  total = 0 # set counter to 

  chars = s.split('')

  chars.each_with_index do | char, i |
    case char
    when "I"
      chars[i + 1] == "V" || chars[i + 1] == "X" ? total -= 1 : total += 1
    when "V"
      total += 5
    when "X"
      chars[i + 1] == "L" || chars[i + 1] == "C" ? total -= 10 : total += 10
    when "L"
      total += 50
    when "C"
      chars[i + 1] == "D" || chars[i + 1] == "M" ? total -= 100 : total += 100
    when "D"
      total += 500
    when "M"
      total += 1000
    end
  end
  return total
end
s = "LVIII"
p roman_to_int(s)

